frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -111,12 +117,19 @@ export function isPipelineSpec(templateString: string) {
 export function getContainer(componentSpec: ComponentSpec, templateString: string) {
   const executionLabel = componentSpec?.executorLabel;
 
-  const jsonTemplate = getPipelineDefFromYaml(templateString);
-  const deploymentSpec = jsonTemplate['deploymentSpec'];
+  const pipelineSpecYAML = getPipelineDefFromYaml(templateString);
+  const pipelineSpec = PipelineSpec.fromJSON(pipelineSpecYAML);
```

Comment:
It looks a bit strange to call `fromJSON()` using a variable called `xxxYAML`. I guess the naming of `pipelineSpecYAML`  can be changed to `pipelineSpecDef`?